docs/articles/2023/20230908-out_of_core_methods.md

```diff
@@ -0,0 +1,133 @@
+* Obtaining batch-corrected highly variable genes. See this [tutorial](../../notebooks/experimental/highly_variable_genes).
+
+These implementations are interwoven with the way users query slices of Census data, which means that these tasks can be seamlessly applied to any slice of the 33M+ cells available in Census.
+
+Continue reading for more implementation details and usage examples.
+
+## Efficient calculation of average and variance gene expression across millions of cells
+
+With `cellxgene_census.experimental.pp.mean_variance` users can now get gene expression average and variance for all genes or cells in a given Census query.
+
+### How it works
+
+Calculations are done in an accumulative chunked fashion, meaning that only a small fraction of the total data is processed at any given time.
+
+The Census data is downloaded in chunks and average and variance accumulators are updated for each chunk. The implementation also takes advantage of CPU-based multiprocessing to speed up the process.
+
+Currently, the mean and variance calculations are done with N being the total number of cells or genes. We are working on enabling calculations where N is then number of non-zero elements.
+
+### Example: *KRAS* and *AQP4* average and variance expression in lung epithelial cells
+
+The following calculates the average and variance values for the genes *KRAS* and *AQP4* in all epithelial cells of the human lung.
+
+Users can easily calculate switch the calculation, and obtain average and variance for each cell across the genes in the query. This is controlled by the `axis` argument of `mean_variance`.
+
+```python
+import cellxgene_census
+import tiledbsoma as soma
+from cellxgene_census.experimental.pp import mean_variance
+
+# Open the Census
+census = cellxgene_census.open_soma()
+human_data = census["census_data"]["homo_sapiens"]
+
+# Set filters
+cell_filter = (
+  "is_primary_data == True "
+  "and tissue_general == 'lung' "
+  "and cell_type == 'epithelial cell'"
+ )
+gene_filter = "feature_name in ['KRAS', 'AQP4']"
+
+# Perform query
+query = human_data.axis_query(
+  measurement_name="RNA",
+  obs_query=soma.AxisQuery(value_filter= cell_filter),
+  var_query=soma.AxisQuery(value_filter= gene_filter)
+)
+
+# Calculate mean and average per gene
+mean_variance_df = mean_variance(query, axis=0, calculate_mean=True, calculate_variance=True)
+
+# Get gene metadata of query
+gene_df = query.var().concat().to_pandas()
+
+query.close()
+census.close()
+```
+
+Which results in:
+
+```python
+mean_variance_df
+# soma_joinid      mean     variance
+# 8624         3.071926  5741.242485
+# 16437        8.233282   452.119153
+
+gene_df
+#    soma_joinid       feature_id feature_name  feature_length
+# 0         8624  ENSG00000171885         AQP4            5943
+# 1        16437  ENSG00000133703         KRAS            6845
+```
```

Comment:
might be nice to join the tables for display purposes, but that also complicates the example just a tad.